.gitattributes

```diff
@@ -1,3 +1,7 @@
 # Ensure Docker script files uses LF to support Docker for Windows.
-setup_docker_prereqs    eol=lf
-/virtualization/Docker/scripts/*    eol=lf
\ No newline at end of file
+# Ensure "git config --global core.autocrlf input" before you clone
+setup_docker_prereqs                text eol=lf
```

Comment:
Shouldn't we just force this for all files instead of pick and choose?

Reply:
It is indeedwhat [github](https://help.github.com/articles/dealing-with-line-endings/) recommends, but then you should EXCLUDE all binary files

Grepping through our repo, it seems that this should work
```
*     text eol=lf
*.py  whitespace=error

*.ico binary
*.jpg binary
*.png binary
*.zip binary
```

Reply:
do we even have ico, jpg, png or zip files in our repo?

Reply:
Indeed

```
Docs/source/static: favicon & png
Tests/resources: zip
Components/camera: jpg
```

Reply:
Let's follow the GitHub recommendations including `* text=auto`

Reply:
Text=auto would try convert line endings to the native format, which is CRLF on Windows and breaks all scripts (running in a docker container or part of docker build).

Originally this was fixed for docker build only (partly if you had autocrlf set to input). 

With this PR I would like to fix all scripts so that I can mount a source tree on my win fs in our dev docker and use scripts.

These days any half decnt editor understand LF, and there should be no issue with our repo only on LF and any windows checkout fully LF

Reply:
So could we do `* text=lf` ?

Reply:
No, it must be `* text eol=lf` according to [git-scm](https://git-scm.com/docs/gitattributes#_checking_out_and_checking_in)

It controls different parts:

* `*`: Applied to all files, hence the need to remove it on binary files, like images&zip later
* `text`: Ensure LF __in the repo__
* `eol=lf`: Attribute to ensure LF __in the working directory__ (This should fix Windows)

Reply:
So confusing. Okay, so  I guess this PR is fine then?

Reply:
Just pushed the version at the top of our discussion, so read to go now

Reply:
Indeed confusing, hopefully we can change it once and just forget about it again.... it only really troubles Windows users